src/java_bytecode/java_types.h

```diff
@@ -18,6 +18,47 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <util/c_types.h>
 #include <util/optional.h>
 
+class java_annotationt : public irept
+{
+public:
+  class valuet : public irept
+  {
+  public:
+    valuet(irep_idt name, exprt value)
+    {
+      id(name);
+      set(ID_value, value);
```

Comment:
I don't know whether you expect these annotations to grow; if you don't, then I'd be more conservative on memory and do `class valuet : public exprt` with `valuet(const irep_idt &name, const exprt &value) : exprt(name) {  copy_to_operands(value); }`, with `const exprt &get_value() { return op0(); }`.